woocommerce/woocommerce
# Introduce woocommerce_order_applied_coupon hook

### All Submissions:






### Changes proposed in this Pull Request:

Add hook for applying coupon to an order. Provides parity with `apply_coupon()` in cart class.
Closes #34473..

### How to test the changes in this Pull Request:

Run this snippet:

```
/**
 * When a coupon is added to an order, add a product.
 * 
 * @param WC_Coupon $coupon
 * @param WC_Order $order
 */
function test_apply_coupon_to_order( $coupon, $order ) {

    $test_product_id = 99; // Adjust to suit.
    
    $test_product = wc_get_product( $test_product_id );
    
    if ( $test_product instanceof WC_Product && $test_product->is_purchasable() ) {
        $order->add_product( $test_product, 1, array( 'total' => 0 ) );
    }
}
add_action( 'woocommerce_order_applied_coupon', 'test_apply_coupon_to_order', 10, 2 );
```

Then create a new order, and add a coupon. Product 99 should now be added to the order (adjust 99 to any simple product ID you have)

### Other information:




### FOR PR REVIEWER ONLY:
